Runtime/Scripts/PersistentConnection.cs

```diff
@@ -0,0 +1,465 @@
+    }
+
+    private void OnApplicationPause(bool pause)
+    {
+      if (pause)
+      {
+        StopAllCoroutines();// stop edge events streaming
+      }
+      else
+      {
+        // resume edge events streaming
+        if (locationUpdatesRunning)
+        {
+          StartCoroutine(OnIntervalEdgeEventsLocation(integration.matchingEngine.EdgeEventsConnection));
+        }
+        if (latencyUpdatesRunning)
+        {
+          StartCoroutine(OnIntervalEdgeEventsLatency(integration.matchingEngine.EdgeEventsConnection, integration.GetHost()));
+        }
+      }
+    }
+
+    private void OnApplicationQuit()
+    {
+      StopAllCoroutines();// stop edge events streaming
+    }
+
+    private void OnDestroy()
+    {
+      StopAllCoroutines();// stop edge events streaming
+    }
+    #endregion
+
+    #region PersistentConnection Functions
+
+    bool ValidateConfigs()
+    {
+      if (config.newFindCloudletEventTriggers.Count == 1) //No FindCloudletTrigger except Error (Added By Default)
+      {
+        PropagateError(FindCloudletEventTrigger.Error, "Missing FindCloudlets Triggers");
+        return false;
+      }
+
+      if (config.latencyThresholdTriggerMs <= 0)
+      {
+        PropagateError(FindCloudletEventTrigger.Error, "latencyThresholdTriggerMs must be greater than 0");
+        return false;
+      }
+
+      if (config.performanceSwitchMargin > 1 || config.performanceSwitchMargin < 0)
+      {
+        PropagateError(FindCloudletEventTrigger.Error, "performanceSwitchMargin must between (0 to 1.0f)");
+        return false;
+      }
+
+      if (config.newFindCloudletEventTriggers.Contains(FindCloudletEventTrigger.LatencyTooHigh) ||
+        config.newFindCloudletEventTriggers.Contains(FindCloudletEventTrigger.CloserCloudlet))
+      {
+        //location
+        if (config.locationConfig.updatePattern == UpdatePattern.OnInterval)
+        {
+          if (config.locationConfig.maxNumberOfUpdates < 0)
+          {
+            PropagateError(FindCloudletEventTrigger.Error, "locationConfig.maxNumberOfUpdates must >= 0");
+            return false;
+          }
+          if (config.locationConfig.updateIntervalSeconds < 0)
+          {
+            PropagateError(FindCloudletEventTrigger.Error, "locationConfig.updateIntervalSeconds must greater >= 0");
+            return false;
+          }
+        }
+        //latency
+        if (config.latencyConfig.updatePattern == UpdatePattern.OnInterval)
+        {
+          if (config.latencyConfig.maxNumberOfUpdates < 0)
+          {
+            PropagateError(FindCloudletEventTrigger.Error, "latencyConfig.maxNumberOfUpdates must greater >= 0");
+            return false;
+          }
+          if (config.latencyConfig.updateIntervalSeconds < 0)
+          {
+            PropagateError(FindCloudletEventTrigger.Error, "latencyConfig.updateIntervalSeconds must greater >= 0");
+            return false;
+          }
+        }
+      }
```

Comment:
@franklin-huang-mobiledgex 
Validation for UpdateConfig when LatencyTooHigh or CloserCloudlet is included among the triggers